app/assets/javascripts/darkswarm/controllers/checkout/checkout_controller.js.coffee

```diff
@@ -20,8 +20,28 @@ Darkswarm.controller "CheckoutCtrl", ($scope, localStorageService, Checkout, Cur
 
   $scope.purchase = (event, form) ->
     event.preventDefault()
+    $scope.formdata = form
     $scope.submitted = true
-    if form.$valid
+
+    if CurrentUser.id
+      $scope.validateForm()
+    else
+      $scope.confirmGuest(true)
+
+  $scope.validateForm = ->
+    if $scope.formdata.$valid
       $scope.Checkout.purchase()
     else
-      $scope.$broadcast 'purchaseFormInvalid', form
+      $scope.$broadcast 'purchaseFormInvalid', $scope.form
+
+  $scope.confirmGuest = ->
```

Comment:
I don't find this method name particularly meaningful since it does nothing to the user. Would it perhaps be better if we just checked by means the HTTP request whether the user exists?